pkg/kubectl/cmd/set/set_image.go

```diff
@@ -210,6 +210,12 @@ func (o *ImageOptions) Run() error {
 			}
 			return nil
 		})
+		if !transformed {
+			if _, err := fmt.Fprintln(o.Out, "no resources changed"); err != nil {
```

Comment:
printing this way messes up output when yaml/json output modes are selected

Reply:
yes, you are right. but this will help add test. we can get output from `cmd.run()`, right?

Reply:
Only when the selected printing mode allows it

Reply:
you mean if spec -o json or -o yaml, we don't print anything? i am ok about this, because seems have user will use `set image -o yaml >> my.yaml` in bash script. this really considerate. i 'd like add this. thanks